torch/csrc/Exceptions.h

```diff
@@ -70,46 +76,30 @@
 
 #define END_HANDLE_TH_ERRORS_PYBIND                                      \
     }                                                                    \
-    catch (py::error_already_set & e) {                                  \
-      /* Unpack already stored error to be detectable by warning code */ \
-      e.restore();                                                       \
-      throw;                                                             \
-    }                                                                    \
-    catch (py::builtin_exception & e) {                                  \
-      /* Unpack already stored error to be detectable by warning code */ \
-      e.set_error();                                                     \
-      throw;                                                             \
-    }                                                                    \
-    catch (torch::jit::JITException & e) {                               \
-      /* Special case for JITException that are explicitly unpacked by */\
-      /* pybind. Set a temporary python error to be detectable by */     \
-      /* warning code */                                                 \
-      PyErr_SetString(PyExc_RuntimeError, "JITException");               \
+    catch(...) {                                                         \
+      __enforce_warning_buffer.set_in_exception();                       \
```

Comment:
Justification for this change: we now only list the special cases in one place and it now handles all exceptions correctly, even edge cases like `cxxabi::__forced_unwind` which don't inherit from `std::exception`.